### PR TITLE
Add sampler.py with Pymbar

### PR DESCRIPTION
**Pull request by chrisjonesBSU**
This PR adds 2 functions:

1. An equilibration sampling function that uses pymbar timeseries to return the uncorrelated samples from an equilibrated region.
2. A function that checks if an array is considered equilibrated given some threshold

Still to do:
- [x] Unit tests
- [ ] Doc strings

### CODE REVIEW

**Comment by chrisjonesBSU** [comment]
Unit tests have been added. Still need to add doc strings.

**Review by jennyfothergill — COMMENTED** [review]
Looks like the test environment needs `pymbar`. I will update the testing worklflow like we did in ~uli~polyBinder, planckton, msibi, etc :)

File: cmeutils/sampling.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        'equilibrated'.
+    conservative : bool, default=True
+        if set to True, uniformly-spaced indices are chosen with interval
+        ceil(g), where g is the statistical inefficiency.  
+        Otherwise, indices are chosen non-uniformly with interval of
+        approximately g in order to end up with approximately T/g total indices
+
+    Returns
+    -------
+    Tuple containing
```
Comment (jennyfothergill) [inline cmeutils/sampling.py:26]:
```suggestion
    (numpy.ndarray, numpy.ndarray, int, int)
        Uncorrelated sample, indices of the uncorrelated values from the input array, index of the production region start from the input array, and the number of effective samples
```

File: cmeutils/sampling.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+    """
+    is_equil, prod_start, ineff, Neff = is_equilibrated(
+            data, threshold_fraction, threshold_neff
+            )
+
+    if is_equil is True:
```
Comment (jennyfothergill) [inline cmeutils/sampling.py:33]: could simply this
```suggestion
    if is_equil:
```

File: cmeutils/sampling.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        uncorr_indices = timeseries.subsampleCorrelatedData(
+                data[prod_start:], g=ineff, conservative=conservative
+            )
+        uncorr_sample = data[prod_start:][uncorr_indices]
+        return(uncorr_sample, uncorr_indices, prod_start, Neff)
+
+    elif is_equil is False:
```
Comment (jennyfothergill) [inline cmeutils/sampling.py:40]:
```suggestion
    else:
```

File: cmeutils/sampling.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    threshold_fraction : float, optional, default=0.8
+        Fraction of data expected to be equilibrated.
+    threshold_neff : int, optional, default=100
+        Minimum amount of effectively correlated samples to consider a_t
+        'equilibrated'.
+    nskip : int, optional, default=1
+        Since the statistical inefficiency is computed for every time origin
+        in a call to timeseries.detectEquilibration, for larger datasets
+        (> few hundred), increasing nskip might speed this up, while
+        discarding more data.
```
Comment (jennyfothergill) [inline cmeutils/sampling.py:76]: might add what is returned

File: cmeutils/sampling.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        return(uncorr_sample, uncorr_indices, prod_start, Neff)
+
+    elif is_equil is False:
+        raise ValueError(
+            "Property does not have requisite threshold of production data "
+            "expected. More production data is needed, or the threshold needs "
+            "to be lowered. See is_equilibrated for more information."
+        )
+
+def is_equilibrated(data, threshold_fraction=0.80, threshold_neff=100, nskip=1):
```
Comment (jennyfothergill) [inline cmeutils/sampling.py:47]: this threshold might be a bit high. In reproducibility study we found we needed to lower it from 0.75 to get reasonable run times.

**Review by jennyfothergill — APPROVED** [review]
LGTM! a couple minor comments

File: cmeutils/sampling.py (hunk 1 above)
Comment (jennyfothergill) [reply cmeutils/sampling.py:26]: I'm not sure what is the best way to document the return type. I've also seen
```
tuple (list, int)
```
so if that seems better, feel free to change it up :)

File: cmeutils/sampling.py (hunk 5 above)
Comment (chrisjonesBSU) [reply cmeutils/sampling.py:47]: Maybe set 0.50 as a default?

**Pull request merged**